Add docs for CCM GCP container costs (#21980)
DataDog/documentation: content/en/cloud_cost_management/container_cost_allocation.md

## Overview

> Datadog Cloud Cost Management (CCM) automatically allocates AWS/Google costs of Kubernetes and AWS ECS clusters to individual pods and tasks running in those clusters. Use cost metrics enriched with tags from pods, nodes, containers, and tasks to visualize container workload cost in the context of your entire cloud bill.

Reviewer: suggestion:
- remove "AWS/Google" from here
- keep "AWS ECS" here
- add "for AWS or Google Cloud" in step 1 below.

Reviewer 2: ```suggestion
Datadog Cloud Cost Management (CCM) automatically allocates costs of Kubernetes and AWS ECS clusters to individual pods and tasks running in those clusters. Use cost metrics enriched with tags from pods, nodes, containers, and tasks to visualize container workload cost in the context of your entire cloud bill.
```

{{< img src="cloud_cost/container_cost_allocation/cost_allocation_table.png" alt="Cloud cost allocation table showing requests and idle costs over the past week" style="width:100%;" >}}

> For Kubernetes clusters, CCM allocates costs of AWS/Google host instances and non-local AWS EBS volumes. For AWS ECS clusters, CCM allocates costs of AWS EC2 instances.

Reviewer 2: ```suggestion
For Kubernetes clusters, CCM allocates costs of AWS and Google host instances, and non-local AWS EBS volumes. For AWS ECS clusters, CCM allocates costs of AWS EC2 instances.
```

## Prerequisites

> 1. Set up and configure the [Cloud Cost integration][1].

Reviewer 2: ```suggestion
1. Set up and configure the [AWS or Google Cloud Cost integration][1].
```

2. For Kubernetes support, install the [**Datadog Agent**][2] in a Kubernetes environment.
     - Ensure that you enable the [**Orchestrator Explorer**][6] in your Agent configuration.
     - Container cost allocation requires **Agent version >= 7.27.0** and **Cluster Agent version >= 1.11.0**.
     - AWS Persistent Volume allocation requires **Agent version >= 7.46.0**

>      - Google supports agentless Kubernetes cost tracking with some limitations, [see below for more details](#agentless-google-kubernetes-costs)

Reviewer: imo we don't need to mention this here in prereqs, the section existing below is enough.

Reviewer 2: ```suggestion
```

3. For AWS ECS support, set up [**Datadog Container Monitoring**][3] in ECS tasks.
     - Optionally, enable [AWS Split Cost Allocation][5] for usage-based ECS allocation.

## Cost allocation

> Cost allocation divides host compute and AWS EBS volume costs in the bill into associated individual tasks or pods. These divided costs are then enriched with tags from nodes, pods, tasks, and volumes (AWS only). This lets you break down costs by any associated dimensions.

Reviewer: I'm not sure if we use this elsewhere or what others think, but "bill" is a bit of a weird term since we ingest Cost Exports not invoices/bills.

```suggestion
Cost allocation divides host compute and AWS EBS volume costs from the cloud provider into associated individual tasks or pods. These divided costs are then enriched with tags from nodes, pods, tasks, and volumes (AWS only). This lets you break down costs by any associated dimensions.
```

Next, Datadog looks at all of the pods running on that node for the day. The cost of the node is allocated to the pod based on the resources it has used and the length of time it ran. This calculated cost is enriched with all of the pod's tags.

**Note**: Only _tags_ from pods and nodes are added to cost metrics. To include labels, enable labels as tags for [nodes][7] and [pods][8].

#### Persistent volume storage (AWS only)

For Kubernetes Persistent Volume storage allocation, Persistent Volumes (PV), Persistent Volume Claims (PVC), nodes, and pods are joined with their associated EBS volume costs. All associated PV, PVC, node, and pod tags are added to the EBS volume cost line items.

Next, Datadog looks at all of the pods that claimed the volume on that day. The cost of the volume is allocated to a pod based on the resources it used and the length of time it ran. These resources
include the provisioned capacity for storage, IOPS, and throughput. This allocated cost is enriched with all of the pod's tags.

> #### Agentless Google Kubernetes costs

Reviewer: for consistency
```suggestion
#### Agentless Kubernetes costs (Google Kubernetes Engine only)
```

> In some cases, you may wish to view the costs of GKE clusters without having Datadog infrastructure monitoring enabled. In these cases CCM supports viewing costs allocated using [GKE cost allocation](https://cloud.google.com/kubernetes-engine/docs/how-to/cost-allocations), with some limitations.

Reviewer: idk but needs some editing, maybe from docs team 🙏 

```suggestion
CCM supports GKE clusters that do not have Datadog infrastructure monitoring enabled, with some limitations. Enable [GKE cost allocation](https://cloud.google.com/kubernetes-engine/docs/how-to/cost-allocations) on unmonitored GKE clusters to access this limited feature set.
```

Reviewer 2: What are the limitations? Can you specify? Otherwise I recommend removing reference to limitations if it's easily identified by the user.
```suggestion
To view the costs of GKE clusters without enabling Datadog infrastructure monitoring, use [GKE cost allocation](https://cloud.google.com/kubernetes-engine/docs/how-to/cost-allocations). Enable GKE cost allocation on unmonitored GKE clusters to access this feature set.
```

Based on resources the task has used, Datadog assigns the appropriate portion of the instance's compute cost to that task. The calculated cost is enriched with all of the task's tags and all of the container tags (except container names) running in the task.

### AWS ECS on Fargate

ECS tasks that run on Fargate are already fully allocated in the CUR. CCM enriches that data by adding out-of-the-box tags and container tags to the AWS Fargate cost.

### Everything else

The information on this page pertains to compute and AWS EBS costs for host instances running Kubernetes pods or ECS tasks. All other costs are given the same value and tags as the source metric, `aws.cost.amortized` or `gcp.cost.amortized`.

## Understanding spend

Using the `allocated_spend_type` tag, you can visualize the spend category associated with your cost at the Kubernetes node, AWS ECS host, AWS EBS volume, or cluster level. Costs are allocated into different spend types:

| Spend type | Description    |
| -----------| -----------    |

> | Managed service fee | Cost of associated fees charged by the provider for managing the cluster, such as EKS/GKE fees. |

Reviewer: ```suggestion
| Managed service fee | Cost of associated fees charged by the provider for managing the cluster, such as EKS or GKE fees. |
```

| Usage | Cost of resources used by workloads. |
| Workload idle | Cost of resources that are reserved and allocated but not used by workloads. |
| Cluster idle | Cost of resources that are not reserved by workloads in a cluster. |

### Compute

>  The cost of a host instance is split into two components: 60% for the CPU and 40% for the memory. Each component is allocated to individual workloads based on their resource reservations and usage.

Reviewer: Should we somehow explain that Google compute costs are sometimes already split, and not by these percentages?
Edit: i see it's below, but might be nice to add something here too.

Reviewer 2: ```suggestion
The cost of a host instance is split into two components: 60% for the CPU and 40% for the memory. Each component is allocated to individual workloads based on their resource reservations and usage.
```

## Cost metrics

When the prerequisites are met, new cost metrics automatically appear.

| Cost Metric                    | Description    |
| ---                                | ----------- |
| `aws.cost.amortized.shared.resources.allocated` | EC2 costs allocated by the CPU & memory used by a pod or ECS task, using a 60:40 split for CPU & memory respectively. Also includes allocated EBS costs. <br> *Based on `aws.cost.amortized`* |
| `aws.cost.net.amortized.shared.resources.allocated` | Net EC2 costs allocated by CPU & memory used by a pod or ECS task, using a 60:40 split for CPU & memory respectively. Also includes allocated EBS costs. <br> *Based on `aws.cost.net.amortized`, if available* |
| `gcp.cost.amortized.shared.resources.allocated` | Compute Engine costs allocated by the CPU & memory used by a pod, using 60:40 split for CPU & memory respectively (when no split is already found in the bill). <br> *Based on `gcp.cost.amortized`* |

These new cost metrics include all of your cloud costs. This allows you to continue visualizing all of your cloud costs at one time, with added visibility into the costs of Kubernetes pods and AWS ECS tasks.

> For example, say you have the tag `team` on a storage bucket, a cloud provider managed database, and Kubernetes pods. You can use these new metrics to group costs by `team`, which includes the costs for all 3.

Reviewer 2: ```suggestion
For example, say you have the tag `team` on a storage bucket, a cloud provider managed database, and Kubernetes pods. You can use these new metrics to group costs by `team`, which includes the costs for all three.
```
